### DISCUSSION TITLE
what happens to moved files ?

### DISCUSSION THREAD

**Discussion by redmoon1945** [artemvlas/veretino#1] [General]
Hi, I wonder what happened if a file is moved to a different sub-directory ? Is it automatically detected by veretino and what happens ?

**Comment by artemvlas** [comment]
Moved files are marked as New and Missing respectively. In order for this to be reliably solved automatically, it is necessary to calculate checksums for New files during parsing job, which can take a lot of time, and is inconvenient. Perhaps in future versions this will be implemented as an additional option...
In fact, I have been puzzled by the problem of renamed (moved) files for a long time, and at the moment, any automatically executed solution that I found seemed to me not reliable enough (in terms of data integrity) or fast enough (in terms of performance).
So in the current version of the program there is still a primitive, but fast and reliable principle: consider a moved file as New.

**Reply by redmoon1945** [reply]
Yes it would be a nice feature. I have a collection of 15000 RAW files (photography) and it is very important for me to know if a file has been corrupted.  Not an easy problem to fix. I'll think about it and come back.

**Reply by artemvlas** [reply]
If you need to rename a subfolder, simply create a Branch first. This will avoid losing a large number of ready-made checksums. If you need to rename some individual files with the obligatory preservation of the current checksums, it is quite acceptable to open the database in any text editor and make changes to the path strings...
And more convenient work with moved objects will probably be implemented sometime in the future, if there is a real need for it.

**Reply by artemvlas** [reply]
To create a new Branch, right-click on a subfolder --> Branch Folder. This action will create a new database in this subfolder with the checksums already stored in the main database (no new calculations needed).  After this action, you can freely copy, move, rename the subfolder as you like in the file system. And check its contents in the same way as with a regular working folder.

**Comment by artemvlas** [comment]
Experimental support for detecting **Moved** (or renamed) files has just been added. If interested, join the testing! (need to build current code)

**Reply by artemvlas** [reply]
You need to enable the corresponding option in the Settings, and when adding new checksums to db, a search will be made for their presence in items marked as Missing. If there is a match, the item will have the **'Moved'** status, and the former missing item will have the **'moved out'** one.

**Reply by 197788** [reply]
Awesome, @artemvlas!

**Reply by artemvlas** [reply]
**UPD (Nov2):** Added the ability to search for Moved items as a separate action, without the need to add new ones.
**UPD (Nov4):** Added the ability to import existing Branches. Can save a lot of time when you need to merge/move/rename a large subfolder.
_Join the testing :)_
